app/controllers/shipments_controller.rb

```diff
@@ -0,0 +1,72 @@
+class ShipmentsController < ApplicationController
+
+  def shipment
+    submission_errors = submission_error_handling(params)
+
+    if submission_errors
+      return submission_errors
+    else
+      origin = ActiveShipping::Location.new(country: params[:origin][:country], state: params[:origin][:state], city: params[:origin][:city], zip: params[:origin][:zip])
+
+      destination = ActiveShipping::Location.new(country: params[:destination][:country], state: params[:destination][:state], city: params[:destination][:city], zip: params[:destination][:zip])
+
+      packages = []
+      if params[:packages].length > 0
+        params[:packages].each do |package|
+          weight = package[:weight].to_i
+          dimensions = package[:dimensions].split(",").map!{|string| string.to_i}
+          pack = ActiveShipping::Package.new(weight, dimensions)
+          packages << pack
+        end
+          @quotes = get_quotes(origin, destination, packages)
+          render :json => @quotes.as_json
+      else
+        render:json => {"error" => "Packages is empty"}, :status => :bad_request
```

Comment:
This line may be part of your issues with error message rendering. There's a space missing between `render` and `:json`, and the interpreter may be interpreting this as `render:` `json` or `render:json`, which might not mean what you think it does.